bundle/config/mutator/resolve_variable_references.go

```diff
@@ -13,15 +14,22 @@ import (
 	"github.com/databricks/cli/libs/dyn/dynvar"
 )
 
+const maxResolutionRounds = 100
```

Comment:
I'm not sure if we really should support this, it looks like all configurations can be refactored into 2 level deep anyways. Maybe warning about more deep nesting instead ?